### PR TITLE
i18n(zh-cn): Update `troubleshooting.mdx`

### PR DESCRIPTION
**Pull request by XindiShang**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- Updated the translations of `zh-cn/guids/troubleshooting.mdx`.

### CODE REVIEW

File: src/content/docs/zh-cn/guides/troubleshooting.mdx [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
-
-**解决方案**：把你的导入和导出语句写在你的组件脚本的顶部。
-
-**状态**：现有限制，正在进行修复。
-
-**不确定这是否是你的问题？**
-检查一下是否有其他人报告过[这个问题](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Transform+failed+with+*+error)!
+下面是你在终端可能会看到的一些常见错误信息，包括它们可能的含义及解决方案。
 
 ### 不能在模块外使用导入语句
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/troubleshooting.mdx:13]: I think we should keep the original English error message, don't you?
```suggestion
### Cannot use import statement outside a module（不能在模块外使用导入语句）
```

File: src/content/docs/zh-cn/guides/troubleshooting.mdx [hunk 2]
```diff
@@ -31,46 +21,53 @@ Astro 提供了几个不同的工具来帮助你排除故障和调试你的代
 **不确定这是你的问题？**
 检查一下是否有其他人报告过[这个问题](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Cannot+use+import+statement)!
 
-### 无法渲染组件
+### `document` (或 `window`) 对象未定义
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/troubleshooting.mdx:24]:
```suggestion
### `document` (or `window`) is not defined（`document` (或 `window`) 对象未定义）
```

File: src/content/docs/zh-cn/guides/troubleshooting.mdx [hunk 3]
```diff
@@ -31,46 +21,53 @@ Astro 提供了几个不同的工具来帮助你排除故障和调试你的代
...
+**解决方案**：定位到代码中调用 `document` 或 `window` 的位置。如果你没有直接调用 `document` 或 `window`，但报错依旧存在，请检查你引用的包是否是在客户端运行的。
+
+- 如果该代码位于 Astro 组件中，将其移动到 frontmatter (代码栅栏)之外的 `<script>` 标签中。这会告诉 Astro 在客户端运行该代码，`document` 和 `window` 在客户端是可用的。
+
+- 如果该代码位于框架组件中，尝试使用生命周期方法（例如 React 中的 [`useEffect()`](https://zh-hans.reactjs.org/docs/hooks-reference.html#useeffect)、Vue 中的 [`onMounted()`](https://cn.vuejs.org/api/composition-api-lifecycle.html#onmounted) 和 Svelte 中的 [`onMount()`](https://svelte.dev/docs#run-time-svelte-onmount) ），在渲染后访问这些对象。你也可以通过添加 [`client:only`](/zh-cn/reference/directives-reference/#clientonly) 指令来阻止组件在服务器上渲染。
 
 **状态**：预期 Astro 行为。
 
-#### 不是这个？
+### 期待默认导出
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/troubleshooting.mdx:40]:
```suggestion
### Expected a default export（期待默认导出）
```

File: src/content/docs/zh-cn/guides/troubleshooting.mdx [hunk 4]
```diff
@@ -31,46 +21,53 @@ Astro 提供了几个不同的工具来帮助你排除故障和调试你的代
...
 
-**不确定这是你的问题？**
-检查一下是否有其他人报告过[这个问题](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Unable+to+render+Component)!
+**解决方案**：尝试寻找你正在导入和渲染的任何组件的错误，并确保它能正常工作。考虑从 [astro.new](https://astro.new) 打开一个 Astro 入门模板，并在最小的 Astro 项目中对组件进行故障排除。
 
-### 预期的默认导出
+**状态**：预期 Astro 行为。
 
-当试图导入或渲染无效组件，或无法正常工作的组件时，会出现这个错误。（该特定消息是因为 Astro 中导入 UI 组件方式造成的）
+### 拒绝执行内联脚本
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/troubleshooting.mdx:48]:
```suggestion
### Refused to execute inline script（拒绝执行内联脚本）
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

**Comment by liruifengv** [comment]
image
image

Some code block need update.

File: src/content/docs/zh-cn/guides/troubleshooting.mdx (hunk 1 above)
Comment (XindiShang) [reply src/content/docs/zh-cn/guides/troubleshooting.mdx:13]: Agreed. It'll be easier to debug. I'll fix that.

Comment (liruifengv) [reply src/content/docs/zh-cn/guides/troubleshooting.mdx:13]: You can use github's commit suggestion feature directly. It works very well.

**Review by liruifengv — APPROVED** [review]
All Chinese LGTM. Awaiting review from @delucis or @Yan-Thomas

**Pull request merged**